cpp/src/Ice/Instance.cpp

```diff
@@ -1521,7 +1521,7 @@ IceInternal::Instance::finishSetup(int& argc, const char* argv[], const Ice::Com
     //
     if (!_referenceFactory->getDefaultRouter())
     {
-        optional<RouterPrx> router{communicator->propertyToProxy("Ice.Default.Router")};
+        optional<RouterPrx> router{communicator->propertyToProxy<RouterPrx>("Ice.Default.Router")};
```

Comment:
Shouldn't we use asigment here? propertyToProxy already returns an optional<T>
```suggestion
        auto router = communicator->propertyToProxy<RouterPrx>("Ice.Default.Router");
```